packages/integrations/useCookies/index.ts

```diff
@@ -0,0 +1,103 @@
+import { ref, getCurrentInstance, watchEffect } from 'vue-demi'
+import Cookie from 'universal-cookie'
+import type { IncomingMessage } from 'http'
+
+type RawCookies = {
+  [name: string]: string
+}
+
+let universalCookie: Cookie | undefined
```

Comment:
I am not sure if we should keep the global variable. I think it's better to hand over the controls to users. They can use the hooks globally to have a global instance or use in components locally which will be disposed when the component unmounted.

Reply:
For example use global `universalCookie` if it was provided before

Reply:
I would prefer those utility functions to be side effects free if possible. Global instance sharing is implicit to users which might cause some confusion.

`createCookies` should not rely on the global side effects as well I think. I am thinking something like this could be better.

#### User defined global state

```ts
// global-utilties.ts
export const cookies = useCookies()

// xxx.vue
import { cookies } from './global-utilties'
```

#### Basic use in components

```ts
setup() {
  const cookies = useCookies() // should always create new UniversalCookie instance
  return { cookies }
}
```

#### Shared instances

`createCookies` returns a new `useCookies` hook that shares the instance.

```ts
export const useMyCookies = createCookies()

// xxx.vue
import { useMyCookies } from './utils'

{
  setup() {
    const cookies = useMyCookies() // should always create new UniversalCookie instance
    return { cookies }
  }
}
```